### PR TITLE
[OGL-1608] feat(etno): add how to cite document attribute

### PR DESCRIPTION
**Pull request by rastislav-chynoransky**
## Summary by CodeRabbit

* **New Features**
  * Added automatic “how to cite” generation for documents, including authors, originators, publication details, optional institution/type/time period, and DOI.
  * Exposed the new `how_to_cite` field in the document show API response.
* **Localization**
  * Added user-facing UI labels for citation components (originator and time period) in English and Slovak, including fallback labeling when the originator person is missing.
* **Tests**
  * Added coverage for citation formatting across locales and edge cases, and for the API response schema.

### CODE REVIEW

File: app-modules/etno/src/Models/Document.php [hunk 1]
```diff
@@ -146,6 +148,22 @@ public static function localityRelations(): array
...
+            subtitle: $this->subtitle,
+            authors: $this->authors->pluck('display_name'),
+            originators: $this->originators->pluck('person.display_name'),
+            publicationDate: $this->publication_date_start,
+            institutionName: $this->institution?->name,
+            type: $this->type,
+            timePeriodStart: $this->time_period_start,
+            timePeriodEnd: $this->time_period_end,
+            doi: $this->doi
+        ));
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app-modules/etno/src/Models/Document.php:165]: Eager-loading is prevented by `Model::shouldBeStrict` in non-production envs

File: app-modules/etno/src/Support/CitationFormatter.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        ?string $subtitle,
+        Collection $authors,
+        Collection $originators,
+        ?\DateTimeInterface $publicationDate,
+        ?string $institutionName,
+        ?ItemType $type,
+        ?\DateTimeInterface $timePeriodStart,
+        ?\DateTimeInterface $timePeriodEnd,
+        ?string $doi,
+    ): ?string {
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app-modules/etno/src/Support/CitationFormatter.php:22]: fixed

File: app-modules/etno/tests/Unit/Support/CitationFormatterTest.php [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        subtitle: null,
+        authors: collect(['Milan Kováč']),
+        originators: collect([]),
+        publicationDate: Carbon::parse('2026-03-15'),
+        institutionName: null,
+        type: ItemType::ResearchReport,
+        timePeriodStart: null,
+        timePeriodEnd: null,
+        doi: null
+    );
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app-modules/etno/tests/Unit/Support/CitationFormatterTest.php:95]: added

File: app-modules/etno/src/Support/CitationFormatter.php [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        if ($timePeriod) {
+            $parts[] = '('.__('etno::ui.citations.time_period').": {$timePeriod}).";
+        }
+
+        if ($doi) {
+            $parts[] = "https://doi.org/{$doi}";
+        }
+
+        return implode(' ', $parts);
+    }
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app-modules/etno/src/Support/CitationFormatter.php:142]: fixed

File: app-modules/etno/src/Models/Document.php [hunk 5]
```diff
@@ -146,6 +148,22 @@ public static function localityRelations(): array
         ];
     }
 
+    public function howToCite(): Attribute
+    {
+        return Attribute::get(fn () => CitationFormatter::format(
+            title: $this->title,
+            subtitle: $this->subtitle,
+            authors: $this->authors->pluck('display_name'),
+            originators: $this->originators->pluck('person.display_name'),
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app-modules/etno/src/Models/Document.php]: TBD

[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app-modules/etno/src/Models/Document.php]: @coderabbitai yes

[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app-modules/etno/src/Models/Document.php]: @coderabbitai create the issue

File: app-modules/etno/src/Support/CitationFormatter.php [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+        if ($formattedTypeLabel) {
+            $parts[] = "({$formattedTypeLabel}).";
+        }
+
+        if ($timePeriod !== '') {
+            $parts[] = '('.__('etno::ui.citations.time_period').": {$timePeriod}).";
+        }
+
+        if ($doi) {
+            $parts[] = "https://doi.org/{$doi}";
```
[the remark this answers is not in the document]
Comment (rastislav-chynoransky) [reply app-modules/etno/src/Support/CitationFormatter.php:58]: `$doi` is supposed to contain the ID only

**Review by mtmn — APPROVED** [review]
